warrior/Framework/ClassUtils/netconf.py

```diff
@@ -349,18 +349,24 @@ def __receive(self):
                         self.__wait_resp.set()
                         self.close()
                         return False
+
                 if len(self.__wait_string) != 0 and self.__wait_string[0]:
                     waitstr = self.__wait_string
                     for notification in self.__notification_list:
                         pNote("Checking notification: "
                               "##{}##".format(notification))
                         match = False
                         xml = etree.fromstring(notification)
-                        temp = xml.xpath(waitstr[0], namespaces=waitstr[1])
-                        if isinstance(temp, bool) and temp:
-                            match = True
-                        elif isinstance(temp, list) and len(temp) > 0:
-                            match = True
+                        #Contains list of xpath
```

Comment:
Is it backward compatible ?  The assumption is it always a list.

Reply:
Hi @santhosh-warrior,
It is backward compatible. The value of waitstr is a tuple , from that tuple we are taking the first element i.e (zero index which is xpath). User can pass single xpath or multiple xpaths ..when we are splitting the output is always list .